src/pkcs11_store.c

```diff
@@ -97,36 +101,39 @@ int wolfPKCS11_Store_Open(int type, CK_ULONG id1, CK_ULONG id2, int read,
     void** store)
 {
     unsigned int i;
-    unsigned int found = 0;
+    int found = -1;
     struct obj_hdr *hdr;
     struct store_object *obj;
 
-    for (i = 1; i < KEYVAULT_MAX_ITEMS; i++) {
+    for (i = 0; i < KEYVAULT_MAX_ITEMS; i++) {
         hdr = (struct obj_hdr*)(vault_base + i * KEYVAULT_OBJ_SIZE);
         if ((type == hdr->type) && (id1 == hdr->token_id) &&
                 (id2 == hdr->object_id)) {
             found = i;
             break;
         }
     }
-    if ((!found) && read) {
+    if ((found == -1) && read) {
         *store = NULL;
         return NOT_AVAILABLE_E;
-    } else if (found) {
-        *store = vault_descriptors[found];
-        obj = vault_descriptors[found];
-        memcpy(&obj->hdr, vault_base + found * KEYVAULT_OBJ_SIZE, sizeof(struct obj_hdr));
+    }
+    obj = XMALLOC(sizeof(struct store_object), NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    if (!obj) {
```

Comment:
Prefer `obj == NULL`... thus not assuming NULL == 0.

Reply:
ditto